src/Tests/README.md

```diff
@@ -0,0 +1,111 @@
+# Tests
+
+ONE-vscode's unittests are based on [mocha](https://mochajs.org/) and [@vscode/test-electron](https://github.com/microsoft/vscode-docs/blob/main/api/working-with-extensions/testing-extension.md).
+- use @vscode/test-electron because ONE-vscode uses vscode api
+  - use Extension Development Host which supports full access to the VS Code API
+- interface of mocha: TDD (BDD to test UX will be introduced in the future)
+- use `assert` of [chai](https://www.chaijs.com/api/assert/)
+
+## structure
+
+Tests are placed in `src/Tests`. The structure follows `src/`.
+```
+src/
+  ...
+  Project/
+    Builder.ts
+    Job.ts
+    ...
+  Utils/
+    Helpers.ts
+    ...
+```
+```
+src/
+  ...
+  Project/
+    Builder.ts
+    Job.ts
+    ...
+  Utils/
+    Helpers.ts
+    ...
+  Tests/
+    ...
+    Project/
+      Builder.test.ts
+      Job.test.ts
+      ...
+    Utils/
+      Helpers.test.ts
+      ...
+```
+
+## scripts
+
+There are three scripts for tests.
+- runTest.ts: the test script
+  - uses the `@vscode/test-electron` API to simplify the process of downloading, unzipping, and launching VS Code with extension test parameters
+  - `--extensionTestsPath` points to the test runner script
+- index.ts: the test runner script
+  - programmatically runs the test suite
+  - exactly does glob() to find tests after adding `hooks` and then, runs them
+- hooks.ts: global hooks
+
+## How to run/debug tests?
+
+```
+ONE-vscode$ npm test
+```
+
+OR
+
+Click F5 or run the `Extension Tests` with breaking points for debug.
+
+# Limitation
+
+According to [the official doc](https://github.com/microsoft/vscode-docs/blob/main/api/working-with-extensions/testing-extension.md#using-insiders-version-for-extension-development), tests cannot run by CLI.
```

Comment:
From the link you shared, I guess this is the limitation of _VSCode Insiders_ version (something like daily version?).

Reply:
First, I read below statements.

---

Because of VS Code's limitation, if you are using VS Code stable release and try to run the integration test on CLI, it will throw an error:

> Running extension tests from the command line is currently only supported if no other instance of Code is running.

In general if you run extension tests from CLI, the version the tests run with cannot be running already. As a workaround, you can run the tests in VS Code Stable and use [VS Code Insiders](https://code.visualstudio.com/insiders/) for development. As long as you are not running the tests from CLI in VS Code Insiders but in VS Code Stable, this setup will work fine.

---

> you are using VS Code stable release and try to run the integration test on CLI
> only supported if no other instance of Code is running
> As a workaround, you can run the tests in VS Code Stable and use VS Code Insiders

AFAIU
- now the integration test on CLI cannot run because no other instance of vscode should be running
- workaround: test with vscode stable + dev with vs code insiders

(Because of my poor english, I'm not sure. 😅)